### PR TITLE
feat(device-type,quantity,chart): quantity color

### PR DESCRIPTION
**Pull request by yarsky-tgz**
* added `color` to the `Quantity` model and routes
 * made `series[].color` nullabe everywhere at charts

### CODE REVIEW

File: src/routes/quantity/add.ts [hunk 1]
```diff
@@ -5,6 +5,7 @@ import { schema as siNumberSchema, SiNumber } from '../../models/si-number';
 interface Request {
   body: {
     name: string;
+    color?: string;
```
Comment (everhardt) [inline src/routes/quantity/add.ts:8]: It's a bit ambiguous for who we provide the interface: if it's for the backend, color is not optional as the default is used. If it's for the front-end, it should indeed be optional. I propose to aim it at the backend: that uses it right now and it only requires typescript front-ends using this version or newer to explicitly specify a color.

File: src/routes/quantity/update.ts [hunk 2]
```diff
@@ -8,6 +8,7 @@ interface Request {
   };
   body: {
     name?: string;
+    color?: string;
```
Comment (everhardt) [inline src/routes/quantity/update.ts:11]: Same 'o

File: src/routes/quantity/add.ts (hunk 1 above)
Comment (everhardt) [reply src/routes/quantity/add.ts:8]: Using the same rationale you might argue that a `.default(..)` in any model (in src/models, which are only used in responses from the server) will never be used, as the Interface provides it correctly as non-optional.

**Review by everhardt — CHANGES_REQUESTED** [review]

File: src/routes/quantity/add.ts (hunk 1 above)
Comment (yarsky-tgz) [reply src/routes/quantity/add.ts]: I was thinking about back compatibility here, but have been mistaken with layer a bit and almost added redundant check to controller :)

default value already handles old clients.

so I've removed `?`

File: src/routes/quantity/update.ts (hunk 2 above)
Comment (yarsky-tgz) [reply src/routes/quantity/update.ts:11]: but this is `update`, each field is optional here, or may be I do not understand you completely ?

In `add` I've removed `?`

Comment (everhardt) [reply src/routes/quantity/update.ts:11]: Sorry, you are absolutely right!

**Review by everhardt — APPROVED** [review]

**Comment by everhardt** [comment]
Released in 16.18.0

**Pull request merged**